src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -293,36 +294,42 @@ public SCM build(@NonNull SCMHead head, @CheckForNull SCMRevision revision) {
         }
         return result;
     }
-    
+
     /**
      * Returns true if the branchName isn't matched by includes or is matched by excludes.
-     * 
+     *
      * @param branchName
      * @return true if branchName is excluded or is not included
      */
     protected boolean isExcluded (String branchName){
       return !Pattern.matches(getPattern(getIncludes()), branchName) || (Pattern.matches(getPattern(getExcludes()), branchName));
     }
-    
+
     /**
-     * Returns the pattern corresponding to the branches containing wildcards. 
-     * 
+     * Returns the pattern corresponding to the branches containing wildcards.
+     *
      * @param branchName
      * @return pattern corresponding to the branches containing wildcards
      */
     private String getPattern(String branches){
       StringBuilder quotedBranches = new StringBuilder();
-      for (String wildcard : branches.split(" ")){
+      for (String term : branches.split(" ")){
         StringBuilder quotedBranch = new StringBuilder();
-        for(String branch : wildcard.split("\\*")){
-          if (wildcard.startsWith("*") || quotedBranches.length()>0) {
-            quotedBranch.append(".*");
-          }
-          quotedBranch.append(Pattern.quote(branch));
-        }
-        if (wildcard.endsWith("*")){
-          quotedBranch.append(".*");
+
+        if (term.startsWith("^")) { // Regex mode, ^ is not a valid character in git names
```

Comment:
Haven't you now injected a regex leading character identifier into the GitSCMSource file which is different than the regex leading character identifier used in the [BranchSpec file](https://github.com/jenkinsci/git-plugin/blob/c0bba641d38759778d2f0cbe04eacaf978cef6eb/src/main/java/hudson/plugins/git/BranchSpec.java#L125)?  The BranchSpec file uses a leading ":" character to indicate that the following text is a regular expression.

Is the BranchSpec logic not being honored in this path through the code?  Could this be taught to use BranchSpec rather than replicating the functionality?